ksqldb-metastore/src/main/java/io/confluent/ksql/metastore/MetaStoreImpl.java

```diff
@@ -203,13 +204,17 @@ public void addSourceReferences(
     }
   }
 
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
Let's try to avoid suppressing warnings about complexity. Instead, we can rewrite `matchedSources` creation with Stream API. Something like:

```java
final List<DataSource> matchedSources = getAllDataSources()
        .values()
        .stream()
        .filter(dataSource -> {
          final boolean nameMatches = dataSource
              .getName()
              .text()
              .equalsIgnoreCase(sourceName.text());
          final boolean sourceTypeMatches = sourceType
              .map(st -> st.equals(dataSource.getDataSourceType()))
              .orElse(true);
          return nameMatches && sourceTypeMatches;
        })
        .sorted(Comparator.comparing(x -> x.getName().text()))
        .collect(Collectors.toList());
```

Note that `DataSource` contains both the name and the type via `.getName()` and `.getDataSourceType()` accordingly, so we can also avoid unnecessary calls to `Objects.requireNonNull(getSource(...))` on lines 227-228.